### PR TITLE
#228/Create an endpoint to get a chatroom by id

### LINKED ISSUE #228: Create a GET endpoint to get a chat room
Part of the user story _As a user, I want to be able to see chat room details_
Details: Create a GET /chat/room/{roomId} endpoint to get a chat room. _This endpoint must be protected._

### PR DESCRIPTION
**Pull request by fizcamposancos**
close #228 #227

### CODE REVIEW

**Review by vdedios — APPROVED** [review]
:ok_hand:

**Review by gbudau — APPROVED** [review]
Thanks! LGTM :+1:

File: transcendence-app/src/chat/chat.controller.ts [hunk 1]
```diff
@@ -96,4 +96,20 @@ export class ChatController {
...
+    description: 'get a chatroom',
+    type: ChatRoom,
+  })
+  @ApiBadRequestResponse({ description: 'Bad Request' })
+  @ApiUnprocessableEntityResponse({ description: 'Unprocessable entity' })
+  async getChatRoomById(@Param('id', ParseUUIDPipe) chatRoomId: string) {
+    const chatroom = await this.chatService.getChatRoomById(chatRoomId);
+
+    if (!chatroom) {
+      throw new UnprocessableEntityException();
```
Comment (gbudau) [inline transcendence-app/src/chat/chat.controller.ts:111]: suggestion: Here we could use the `NotFoundException()`

Comment (fizcamposancos) [reply transcendence-app/src/chat/chat.controller.ts]: Done

**Pull request merged**